client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,388 @@
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  > svg {
+    width: 100%;
+    height: 100%;
+    > path {
+      fill: ${prop('accentColor')};
+      stroke: ${prop('accentColor')};
+    }
+  }
+`;
+
+const Title = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${remSize(2)};
+
+  > * {
+    padding: 0;
+    margin: 0;
+  }
+
+  > h5 {
+    font-size: ${remSize(13)};
+    font-weight: normal;
+  }
+`;
+
+const Options = styled.div`
+  margin-left: auto;
+  display: flex;
+  /* transform: translateX(${remSize(12)}); */
+  svg {
+    fill: ${(props) => props.color};
+  }
+
+  /* Drop Down menu */
+
+  > div > button:focus + ul,
+  > div > ul > button:focus ~ div > ul {
+    transform: scale(1);
+    opacity: 1;
+  }
+
+  > div {
+    position: relative;
+    ul {
+      ${prop('MobilePanel.default')}
+      position: absolute;
+      overflow: hidden;
+      z-index: 10;
+      right: 10px;
+      transform: translateX(${remSize(6)});
+      width: max-content;
+      transform: scale(0);
+      opacity: 0;
+      transform-origin: top;
+      transition: opacity 100ms;
+      transition: transform 100ms 200ms;
+      display: flex;
+      flex-direction: column;
+      gap: ${remSize(2)};
+      box-shadow: rgba(60, 64, 67, 0.3) 0px 1px 2px 0px,
+        rgba(60, 64, 67, 0.15) 0px 1px 3px 1px;
+      min-width: ${remSize(120)};
+      border-radius: ${remSize(2)};
+
+      /* User */
+      li.user {
+        color: ${prop('colors.processingBlue')};
+        font-size: ${remSize(16)};
+        font-weight: bold;
+        margin: ${remSize(6)} ${remSize(12)};
+      }
+
+      > b {
+        margin: ${remSize(6)} ${remSize(12)};
+        position: relative;
+        width: min-content;
+        &::after {
+          content: '';
+          position: absolute;
+          opacity: calc(0.5);
+          top: 50%;
+          transform: translate(100%, -50%);
+          right: ${remSize(-6)};
+          width: ${remSize(55)};
+          height: 1px;
+          background-color: ${prop('Button.primary.default.foreground')};
+        }
+      }
+
+      > li {
+        display: flex;
+        width: 100%;
+        > button {
+          width: 100%;
+          padding: ${remSize(8)} ${remSize(15)} ${remSize(8)} ${remSize(10)};
+          font-size: ${remSize(18)};
+          text-align: left;
+          &:hover {
+            background-color: ${prop('Button.primary.hover.background')};
+            color: ${prop('Button.primary.hover.foreground')};
+          }
+        }
+      }
+    }
+  }
+`;
+
+const MobileNav = (props) => {
+  const project = useSelector((state) => state.project);
+  const user = useSelector((state) => state.user);
+  const [title, setTitle] = useState();
+
+  // use the useNavigate hook in the react router v6
+  const history = useHistory();
+  const navigate = (url) => history.push(url);
+
+  const { pathname } = useLocation();
+  const editorLink = useSelector(selectSketchPath);
```

Comment:
Isn’t it nice when we have defined selector functions?  Makes it much easier to access the same state in multiple places.

Reply:
yes it is ✨